packages/core/helper-plugin/lib/src/components/GenericInput/tests/index.test.js

```diff
@@ -43,6 +44,29 @@ function setupNumber(props) {
   };
 }
 
+function setupDatetimePicker(props) {
+  const DATETIMEPICKER_FIXTURE_PROPS = {
+    type: 'datetime',
+    name: 'datetime-picker',
+    intlLabel: {
+      id: 'label.test',
+      defaultMessage: 'datetime picker',
+    },
+    value: null,
+    onChange: jest.fn(),
+    onClear: jest.fn(),
+    ...props,
+  };
+
+  const rendered = render(<ComponentFixture {...DATETIMEPICKER_FIXTURE_PROPS} />);
+
+  return {
+    ...rendered,
+  };
+}
+
+jest.setTimeout(50000);
```

Comment:
Are we sure this doesn't set it globally? :)

Reply:
Why do we not set it on the test directly? `test('name', testFunc, timeout)` is the signature. I think then jest takes care to reset it afterwards to the default value.

Reply:
*50s

Reply:
Also we should be writing comments as to why we've changed the timeout, other developers won't know

Reply:
Yesterday I tried different attempts in the previous PR, https://github.com/strapi/strapi/pull/16338 use beforeAll and afterAll to set the timeout, define the timeout inside the test signature but no one of these methods worked, then I searched inside the project a similar situation and I found two tests setting the timeout to 50s like I did, https://github.com/strapi/strapi/blob/b8f22d3d669c444fabaf9cf44bc2d9dd625d6b92/packages/core/admin/admin/src/pages/MarketplacePage/tests/plugins.test.js#L14 and this one https://github.com/strapi/strapi/blob/b8f22d3d669c444fabaf9cf44bc2d9dd625d6b92/packages/core/admin/admin/src/pages/MarketplacePage/tests/providers.test.js#L16 I am ok to add a comment, I don't think my instruction set the timeout globally, because it is already used inside the code.